pkg/deploy/graph/analysis/dc.go

```diff
@@ -124,3 +131,89 @@ Node:
 
 	return markers
 }
+
+func FindPersistentVolumeClaimWarnings(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, uncastDcNode := range g.NodesByKind(deploygraph.DeploymentConfigNodeKind) {
+		dcNode := uncastDcNode.(*deploygraph.DeploymentConfigNode)
+		marker := pvcMarker(g, f, dcNode)
+		if marker != nil {
+			markers = append(markers, *marker)
+		}
+	}
+
+	return markers
+}
+
+func pvcMarker(g osgraph.Graph, f osgraph.Namer, dcNode *deploygraph.DeploymentConfigNode) *osgraph.Marker {
+	for _, uncastPvcNode := range g.SuccessorNodesByEdgeKind(dcNode, deployedges.VolumeClaimEdgeKind) {
+		pvcNode := uncastPvcNode.(*kubegraph.PersistentVolumeClaimNode)
+
+		if !pvcNode.Found() {
+			return &osgraph.Marker{
+				Node:         dcNode,
+				RelatedNodes: []graph.Node{uncastPvcNode},
+
+				Severity: osgraph.WarningSeverity,
+				Key:      MissingPVCWarning,
+				Message:  fmt.Sprintf("%s points to a missing persistent volume claim (%s).", f.ResourceName(dcNode), f.ResourceName(pvcNode)),
+				// Suggestion: osgraph.Suggestion(fmt.Sprintf("oc create pvc ...")),
+			}
+		}
+
+		onSingleHost := len(g.NodesByKind(kubegraph.NodeKind)) == 1
+		isRollingDeployment := dcNode.DeploymentConfig.Spec.Strategy.RollingParams != nil
+
+		// If the claim is not RWO or the cluster has only one node, then rolling deployments
+		// should work just fine.
+		if !hasRWOAccess(pvcNode) || onSingleHost || !isRollingDeployment {
+			continue
+		}
+
+		marker := &osgraph.Marker{
+			Node:         dcNode,
+			RelatedNodes: []graph.Node{uncastPvcNode},
+
+			Severity: osgraph.WarningSeverity,
+			Key:      SingleHostVolumeWarning,
+			Message:  fmt.Sprintf("%s may block during deployments due to its RWO volume.", f.ResourceName(dcNode)),
```

Comment:
ok

I've separated between three  cases:
1) dc.spec.replicas = 1 && maxSurge > 0 ---> the next deployment will be blocked, warning
2) dc.spec.replicas > 0 ---> the deployment is already blocked, error
3) otherwise, dc.spec.replicas = 1, maxSurge = 0 ---> warning